src/Admin/PageAdmin.php

```diff
@@ -62,6 +62,7 @@ public function __construct(
     protected function configureRoutes(RouteCollectionInterface $collection): void
     {
         $collection->add('compose', $this->getRouterIdParameter().'/compose');
+        //TODO how is it linked with src/Controller/PageAdminController::composeContainerShowAction, where the route is defined?
```

Comment:
do you have any idea where this router is defined? `compose_container_show`

Other question: WDYT to  extract this [composeContainerShowAction](https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/Controller/PageAdminController.php#L257) for other controller like `PageAdminComposeContainerShowController` or just `PageAdminComposeController` with `composeContainerShowAction` and `composerAction`?

Reply:
I would say It's just linked with the url you gave `compose/container/{id}`

Reply:
hmmm I see, but do you have any idea where the route is defined for https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/Controller/PageAdminController.php#L257 👀

Reply:
Here: https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/Admin/PageAdmin.php#L65

Reply:
The first parameter gets converted camelCase and added "Action" at the end, and that's how it knows the name.

Reply:
I think it's done with the name instead.

when you call `add->('compose_container_show')` the RouteCollection/RouteGenerator are automatically looking for a `composeContainerShow`
https://docs.sonata-project.org/projects/SonataAdminBundle/en/4.x/cookbook/recipe_custom_action/

Reply:
> I think it's done with the name instead.
> 
> when you call `add->('compose_container_show')` the RouteCollection/RouteGenerator are automatically looking for a `composeContainerShow` https://docs.sonata-project.org/projects/SonataAdminBundle/en/4.x/cookbook/recipe_custom_action/

hmmmm I guess I got it :D, Thanks :)